src/app/shared/data-input/data-input.component.html

```diff
@@ -0,0 +1,49 @@
+<ng-container *ngIf="constraint; else textDataInput"
+              [ngSwitch]="constraint.type">
+  <boolean-data-input *ngSwitchCase="'Boolean'"
```

Comment:
what about defining readonly properties in component and then use it in switch cases?
`public readonly booleanType = ConstraintType.Boolean` etc...

Reply:
Thanks, I will change it. But there is even better solution:
```typescript
public readonly constraintType = ConstraintType;
```
...and then we will have something like `*ngSwitchCase="constraintType.Boolean"` in the template.